### PR TITLE
test: miscellaneous fixes

### PR DESCRIPTION
**Pull request by ianvernon**
- test/runtime: create and delete containers before policy tests
    - This ensures that Cilium can handle invalid policy while endpoints are running, and that cleanup tests are tested with endpoints running.
- test/helpers: always log current state of endpoints
    - Always log the number of "ready" endpoints to the logger in WaitEndpointsReady.

Signed-off by: Ian Vernon <ian@cilium.io>

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

File: test/runtime/Policies.go
Comment (joestringer) [inline test/runtime/Policies.go:692]: Whenever we instantiate these containers, I think we want to know that they're ready when it's returned. It'd be nice to fold in these waits (and maybe even the `Expect()`) into the `SampleContainersActions()` function. Maybe by using `gomega.ExpectWithOffset(...)` in the helper function so that ginkgo points out the correct line in the actual test file where it failed.

**Review by joestringer — APPROVED** [review]
LGTM, one minor note we could fix up later.

File: test/runtime/Policies.go
Comment (ianvernon) [reply test/runtime/Policies.go:692]: This is a good idea, but in the interest of making sure that the specified tests are testing with endpoints running, I'm going to merge this after getting other CI code-owner approval.

**Review by aanm — APPROVED** [review]

**Pull request merged**
